internal/completions/httpapi/handler.go

```diff
@@ -92,6 +93,14 @@ func newCompletionsHandler(
 			return
 		}
 
+		isDotcom := dotcom.SourcegraphDotComMode()
+		if !isDotcom {
+			err := checkClientCodyIgnoreCompatibility(r)
+			if err != nil {
```

Comment:
Stylistic nit-pick. This is arguably cleaner, since the name of the function is relatively short. But please reject if you do not think this is strictly an improvement.

```suggestion
			if err := checkClientCodyIgnoreCompatibility(r); err != nil {
```